src/overlays/actors/ovl_Obj_Mure2/z_obj_mure2.c

```diff
@@ -30,43 +47,209 @@ ActorInit Obj_Mure2_InitVars = {
     /**/ NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_809615F4[] = {
-    ICHAIN_F32(uncullZoneForward, 100, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneScale, 2100, ICHAIN_CONTINUE),
-    ICHAIN_F32(uncullZoneDownward, 100, ICHAIN_STOP),
+static f32 sActivationRangesSq[] = {
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_BUSH_RING
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    SQ(1600.0f), // OBJMURE2_CHILDTYPE_ROCK_RING
 };
 
-#endif
-
-extern InitChainEntry D_809615F4[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960CF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960E0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80960F0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_80961018.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_8096104C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809611BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809612BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/ObjMure2_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613C4.s")
+static f32 sDeactivationRangesSq[] = {
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_BUSH_RING
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    SQ(1705.0f), // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613E8.s")
+static s16 sChildCounts[] = {
+    9,  // OBJMURE2_CHILDTYPE_BUSH_RING
+    12, // OBJMURE2_CHILDTYPE_BUSH_SCATTERED
+    8,  // OBJMURE2_CHILDTYPE_ROCK_RING
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Mure2/func_809613FC.s")
+static s16 sActorIds[] = {
+    ACTOR_EN_KUSA,
+    ACTOR_EN_KUSA,
+    ACTOR_EN_ISHI,
+};
+typedef struct Mure2SpawnInfo {
+    s16 dist;
+    s16 angle;
+} Mure2SpawnInfo;
```

Comment:
I should point out that there is a VecPolar type I added to z64math.h in my grass documentation pr that takes a float for distance instead of a s16